src/server/engine_shard_set.cc

```diff
@@ -265,7 +265,13 @@ bool EngineShard::DefragTaskState::CheckRequired() {
   return false;
 }
 
-bool EngineShard::DoDefrag() {
+void EngineShard::ForceDefrag() {
+  DefragTaskState defrag_state_;
+  while (DoDefrag(defrag_state_))
```

Comment:
this won't do. it blocks the entire server. 
please note that you only need it run once (to start it) and then CheckRequired will continue running it in background due to `cursor > 0`

Reply:
thx, it looks I misunderstand the task a little bit